src/poetry/puzzle/provider.py

```diff
@@ -211,12 +212,13 @@ def get_package_from_vcs(
         branch: Optional[str] = None,
         tag: Optional[str] = None,
         rev: Optional[str] = None,
+        directory: Optional[str] = None,
         name: Optional[str] = None,
     ) -> "Package":
         if vcs != "git":
             raise ValueError(f"Unsupported VCS dependency {vcs}")
 
-        suffix = url.split("/")[-1].rstrip(".git")
+        suffix = url.split("/")[-1][:-4]
```

Comment:
`rstrip` removes trailing instances of characters not the string itself. So if you had repos that had names containing g,i,t it would be removed. For example, `gotit.git` would become `go`

Reply:
Isn't the url guaranteed to end in `.git` in this function? Currently poetry only supports git as vcs. Or is this to allow for non git vcs down the line?

Reply:
I don't think so (but I might be wrong!); Github happens to attach a `.git` to the names of all their fetch endpoints but I don't think that's actually part of the git protocol. You can expose a git repo with any valid URL. You can see from other VCS providers like Sourcehut, where the URL used to pull git data is that same as the http endpoint (eg. there's no added `.git` suffix). Example: https://git.sr.ht/~sircmpwn/aerc

Reply:
Ah I see. Thanks for pointing that out. Pushed the fix